ethcore/src/state/mod.rs

```diff
@@ -2196,6 +2266,27 @@ mod tests {
 		assert_eq!(state.balance(&a).unwrap(), U256::from(0));
 	}
 
+	#[test]
+	fn checkpoint_get_storage_at() {
+		let mut state = get_temp_state();
+		let a = Address::zero();
+		let k = H256::from(U256::from(0));
+		let c1 = state.checkpoint();
+		state.set_storage(&a, k, H256::from(U256::from(1))).unwrap();
+		let c2 = state.checkpoint();
+		let c3 = state.checkpoint();
+		state.set_storage(&a, k, H256::from(U256::from(3))).unwrap();
+		let c4 = state.checkpoint();
+		state.set_storage(&a, k, H256::from(U256::from(4))).unwrap();
+		let c5 = state.checkpoint();
+
+		assert_eq!(state.checkpoint_storage_at(c1, &a, &k).unwrap(), Some(H256::from(U256::from(0))));
+		assert_eq!(state.checkpoint_storage_at(c2, &a, &k).unwrap(), Some(H256::from(U256::from(1))));
```

Comment:
Why is it `1`? If we discard all checkpoints `c3,c4,c5` I believe we should be back at `0`?
Also its a bit weird that checking `checkpoint_storage_at(c2)` before creating checkpoint `c3` would return different values, but perhaps i'm missing something.

Reply:
The `c2` checkpoint is created after `state.set_storage(1)`. So in that case at `c2` the value should be `1`?